platform/darwin/src/http_file_source.mm

```diff
@@ -91,8 +91,10 @@ void cancel() {
 
             NSURLSessionConfiguration *sessionConfig =
             [MGLNetworkConfiguration sharedManager].sessionConfiguration;
-            
-            session = [NSURLSession sessionWithConfiguration:sessionConfig];
+            id<NSURLSessionDelegate> sessionDelegate =
+            [MGLNetworkConfiguration sharedManager].sessionDelegate;
+
+            session = [NSURLSession sessionWithConfiguration:sessionConfig delegate:sessionDelegate delegateQueue:nil];
```

Comment:
Since [`sessionWithConfiguration:delegate:delegateQueue:`](https://developer.apple.com/documentation/foundation/nsurlsession/1411597-sessionwithconfiguration) keeps a strong reference to the delegate, I think we'll need to call ` invalidateAndCancel` or `finishTasksAndInvalidate` in the destructor.

Reply:
Sounds good. I can add a destructor to `HTTPFileSource::Impl` and call `invalidateAndCancel` on the`NSURLSession` there, but how can I test it? Is there a way to tear down the stack? I have tried to stop the app, but the destructor does not seem to be called.

Reply:
Apologies for the delay in responding - I believe we'll need to create a test for this. There are cpp tests for `HTTPFileSource` and `OnlineFileSource` which we might be able to take advantage of, but I think we'll need to create an integration test for this.

Reply:
cc: @tmpsantos for an interesting testing scenario (platform <-> core).

Reply:
Added: https://github.com/mapbox/mapbox-gl-native/issues/15592